src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3309,15 +3342,179 @@ impl Coordinator {
 
         // Execute the `optimize/finalize_dataflow` stage.
         let dataflow_plan = catch_unwind(broken, "finalize_dataflow", || {
-            self.finalize_dataflow(dataflow, cluster_id)
+            self.finalize_dataflow(dataflow, target_cluster_id)
         })?;
 
         // Trace the resulting plan for the top-level `optimize` path.
         trace_plan(&dataflow_plan);
 
         // Return objects that need to be passed to the `ExplainContext`
         // when rendering explanations for the various trace entries.
-        Ok((used_indexes, fast_path_plan, dataflow_metainfo))
+        Ok((
+            used_indexes,
+            fast_path_plan,
+            dataflow_metainfo,
+            BTreeMap::new(),
+        ))
+    }
+
+    #[tracing::instrument(target = "optimizer", level = "trace", name = "optimize", skip_all)]
+    async fn explain_create_materialized_view_optimizer_pipeline(
+        &mut self,
+        name: QualifiedItemName,
+        raw_plan: mz_sql::plan::HirRelationExpr,
+        column_names: Vec<ColumnName>,
+        target_cluster_id: ClusterId,
+        broken: bool,
+    ) -> Result<
+        (
+            UsedIndexes,
+            Option<FastPathPlan>,
+            DataflowMetainfo,
+            BTreeMap<GlobalId, TransientItem>,
+        ),
+        AdapterError,
+    > {
+        use mz_repr::explain::trace_plan;
+
+        if broken {
+            tracing::warn!("EXPLAIN ... BROKEN <query> is known to leak memory, use with caution");
+        }
```

Comment:
Do we have a feature flag for this? If not we should add one.

Reply:
Yes, `enable_explain_broken`.